src/components/Forms/AddToCartForm/AddToCartForm.test.js

```diff
@@ -0,0 +1,102 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import AddToCartForm from './AddToCartForm';
+
+jest.mock('react-redux');
+
+describe('AddToCartForm', () => {
+  const labelName = 'Product name';
+  const labelImg = 'Product image url';
+  const labelPrice = 'Product price';
+  const invalidNameMsg = 'The name should be longer';
+  const invalidImgMsg = 'The url should lead to the image';
+  const invalidPriceMsg = 'The price should be numeric';
+
+  it('renders empty form', () => {
+    const dispatch = jest.fn();
+    useDispatch.mockReturnValue(dispatch);
+    const { getByRole, getByTestId } = render(<AddToCartForm />);
+    const productNameInput = getByRole('textbox', { name: labelName });
+    const productImageInput = getByRole('textbox', {
+      name: labelImg,
+    });
+    const productPriceInput = getByRole('spinbutton', {
+      name: labelPrice,
+    });
+    const createProductBtn = getByTestId('create-product-btn');
+
+    expect(productNameInput).toHaveDisplayValue('');
+    expect(productImageInput).toHaveDisplayValue('');
+    expect(productPriceInput).toHaveDisplayValue('');
+    expect(createProductBtn).toBeDisabled();
+  });
+
+  it('checks product name input valid', () => {
+    const dispatch = jest.fn();
+    useDispatch.mockReturnValue(dispatch);
+    const { getByRole, getByText, queryByText } = render(<AddToCartForm />);
+    const productNameInput = getByRole('textbox', { name: labelName });
+
+    userEvent.type(productNameInput, 'new');
+    expect(getByText(invalidNameMsg)).toBeInTheDocument();
+    userEvent.type(productNameInput, 'new product');
+    expect(queryByText(invalidNameMsg)).toBeNull();
+  });
+
+  it('checks product image input valid', () => {
+    const dispatch = jest.fn();
+    useDispatch.mockReturnValue(dispatch);
+    const { getByRole, getByText, queryByText } = render(<AddToCartForm />);
+    const productImageInput = getByRole('textbox', {
+      name: labelImg,
+    });
+
+    userEvent.type(productImageInput, 'https://image');
+    expect(getByText(invalidImgMsg)).toBeInTheDocument();
+    userEvent.type(productImageInput, 'https://image.png');
```

Comment:
Probably this URL is incorrect too :)
Did you use `image.png` as a domain name or as a file name?

Reply:
I used the url as a domain name, solved https://github.com/maxSedelnikov/react-shopping-app/pull/6/commits/c759e7206dd9006634852446e70ed64047e50a1d